sdk/spring/scripts/compatibility_insert_dependencymanagement.py

```diff
@@ -120,12 +108,45 @@ def add_dependency_management_for_file(file_path, spring_boot_dependencies_versi
             repo_content = pom_file_content[:insert_position] + insert_content + pom_file_content[insert_position:]
             with open(file_path, 'r+', encoding = 'utf-8') as updated_pom_file:
                 updated_pom_file.writelines(repo_content)
+    with open(file_path, 'r', encoding = 'utf-8') as pom_file:
+        pom_file_content = pom_file.read()
+        insert_position = get_dependency_management_position(pom_file_content)
+        insert_content = get_dependency_management_content(pom_file_content)
+        dependency_content = pom_file_content[:insert_position] + insert_content + pom_file_content[insert_position:]
+        insert_position = get_prop_position(pom_file_content)
+        insert_content = get_prop_content(pom_file_content, spring_boot_dependencies_version, spring_cloud_dependencies_version)
+        prop_content = dependency_content[:insert_position] + insert_content + dependency_content[insert_position:]
+        with open(file_path, 'r+', encoding = 'utf-8') as updated_pom_file:
+            updated_pom_file.writelines(prop_content)
 
 
-def get_dependency_management_content():
-    return """
+def contains_dependency_management(pom_file_content):
+    return pom_file_content.find("<dependencyManagement>") != -1
+
+
+def get_dependency_management_position(pom_file_content):
+    if contains_dependency_management(pom_file_content):
+        return pom_file_content.find("</dependencies>")
+    else:
+        return pom_file_content.find("<dependencies>")
+
+
+def get_dependency_management_content(pom_file_content):
+    if contains_dependency_management(pom_file_content):
+        return get_dependency_management_content_without_tag()
+    else:
+        return """
   <dependencyManagement>
     <dependencies>
+      {}
+    </dependencies>
+  </dependencyManagement>
+  
+""".format(get_dependency_management_content_without_tag())
```

Comment:
what if the `<dependencyManagement>` already contains a `spring-boot-dependencies` or `spring-cloud-dependencies`?